src/ripple/app/misc/TxQ.h

```diff
@@ -267,18 +402,36 @@ class TxQ
             };
         }
 
+        /** Use the number of transactions in the current open ledger
+            to compute the fee level a transaction must pay bypass the
+            queue.
+
+            @param view Current open ledger.
+            @param txCountPadding Optional number of "extra" transactions
+                to assume are in the ledger. Can be used to determine a
+                padded fee a transaction can pay to beat races with other
+                transactions.
+
+            @return A fee level value.
+        */
         static
         std::uint64_t
         scaleFeeLevel(Snapshot const& snapshot, OpenView const& view,
             std::uint32_t txCountPadding = 0);
 
         /**
-            Returns the total fee level for all transactions in a series.
-            Assumes that there are already more than txnsExpected_ txns in
-            the view. If there aren't, the math still works, but the level
-            will be high.
-
-            Returns: A `std::pair` as returned from `mulDiv` indicating whether
+            Computes the total fee level for all transactions in a series.
+            Assumes that there are already more than @ref txnsExpected_ txns
+            in the view. If there aren't, the math still works, but the level
+            will be higher than actually required.
+
+            @param view Current open / working ledger. (May be a sandbox.)
+            @param extraCount Number of additional transactions to count as
+                in the ledger.
+            @param seriesSize Total number of transactions in the series to be
+                processed.
+
+            @return A `std::pair` as returned from `mulDiv` indicating whether
```

Comment:
That's why I reference `mulDiv`. https://github.com/ximinez/rippled/blob/txq_doc/src/ripple/basics/mulDiv.h#L29

The `bool` can be `false` if the calculation overflows. The reason it returns a `pair` instead of an `optional` is that it populates a maxint value for the result, which the caller can use anyways if it wants.

Reply:
Hmmm.  Fair.  If I'm desperate enough I can go look up the documentation on `mulDiv` at the cost of some small additional annoyance 😉 .

Reply:
I did update the description slightly, to mention that the flag indicates overflow.